WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -0,0 +1,1047 @@
+        let width = view.frame.width
+        tableViewHandler.refreshCachedRowHeightsForWidth(width)
+    }
+    
+    override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        
+        needsRefreshCachedCellHeightsBeforeLayout = true
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        
+        if needsRefreshCachedCellHeightsBeforeLayout {
+            needsRefreshCachedCellHeightsBeforeLayout = false
+            
+            let width = view.frame.width
+            
+            tableViewHandler.refreshCachedRowHeightsForWidth(width)
+            
+            if let indexPathsForVisibleRows = tableView.indexPathsForVisibleRows {
+                tableView.reloadRowsAtIndexPaths(indexPathsForVisibleRows, withRowAnimation: .None)
+            }
+        }
+    }
+    
+    // MARK: - Multitasking Support
+    
+    func handleApplicationDidBecomeActive(notification: NSNotification) {
+        needsRefreshCachedCellHeightsBeforeLayout = true
+    }
+    
+    // MARK: - Configuration
+    
+    
+    func heightForFooterView() -> CGFloat
+    {
+        return self.dynamicType.defaultHeightForFooterView
+    }
+
+    
+    override func preferredStatusBarStyle() -> UIStatusBarStyle {
+        return .LightContent
+    }
+    
+    func configureNavbar() {
+        // IMPORTANT: this code makes sure that the back button in WPPostViewController doesn't show
+        // this VC's title.
+        //
+        let backButton = UIBarButtonItem(title: "", style: .Plain, target: nil, action: nil)
+        navigationItem.backBarButtonItem = backButton
+        
+        let rightBarButtonItem = UIBarButtonItem(customView: rightBarButtonView)
+        WPStyleGuide.setRightBarButtonItemWithCorrectSpacing(rightBarButtonItem, forNavigationItem:navigationItem)
+        
+        navigationItem.titleView = filterButton
+        updateFilterTitle()
+    }
+    
+    func configureCellsForLayout() {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func configureTableView() {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func configureFooterView() {
+        
+        let mainBundle = NSBundle.mainBundle()
+        
+        guard let footerView = mainBundle.loadNibNamed("PostListFooterView", owner: nil, options: nil)[0] as? PostListFooterView else {
+            preconditionFailure("Could not load the footer view from the nib file.")
+        }
+        
+        postListFooterView = footerView
+        postListFooterView.showSpinner(false)
+        
+        var frame = postListFooterView.frame
+        frame.size.height = heightForFooterView()
+        
+        postListFooterView.frame = frame
+        tableView.tableFooterView = postListFooterView
+    }
+    
+    func getNoResultsView() -> WPNoResultsView {
```

Comment:
You could use a lazy initialiser instead of this method, and avoid all the checking for nil.